### ISSUE TITLE
False positive warning for CA1508 ("Avoid dead conditional code")

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn-analyzers#5329]
**Version Used**: 
dotnet 5.0.301
vscode

**Steps to Reproduce**:

I receive an input that is allowed to be null, but is not allowed to be empty nor whitespace.

````cs
var s = "  \t";
if (s != null && string.IsNullOrWhiteSpace(s)) {    // <-- warning CA1508
  // ...
}
````

Warning:
> 'string.IsNullOrWhiteSpace(s)' is always 'false'. Remove or refactor the condition(s) to avoid dead code.

**Expected Behavior**:
If it is known that the string is not null, then the "empty or whitespace" behavior of `IsNullOrWhiteSpace` still applies. The warning should not occur.

**Comment by DmitryKondratenko1986** [comment]
I opened the same issue two days before [https://github.com/dotnet/roslyn-analyzers/issues/5305]()

**Comment by lonix1** [comment]
I found a workaround, which is better than the code above:

````cs
if (s?.Trim().Length == 0) {
}
````

However the false positive for `IsNullOrWhiteSpace()` still exists.

**Comment by mavasani** [comment]
Dupe of #5329
